qiskit_nature/transformers/active_space_transformer.py

```diff
@@ -0,0 +1,193 @@
+                             only be used to enforce an active space that is not chosen purely
+                             around the Fermi level.
+        """
+        self.num_electrons = num_electrons
+        self.num_orbitals = num_orbitals
+        self.num_alpha = num_alpha
+        self.active_orbitals = active_orbitals
+
+    def transform(self, q_molecule: QMolecule) -> QMolecule:
+        """Reduces the given `QMolecule` to a given active space.
+
+        Args:
+            q_molecule: the `QMolecule` to be transformed.
+
+        Returns:
+            A new `QMolecule` instance.
+
+        Raises:
+            QiskitNatureError: If more electrons or orbitals are requested than are available, if an
+                               uneven number of inactive electrons remains, or if the number of
+                               selected active orbital indices does not match `num_orbitals`.
+        """
+        # get molecular orbital coefficients
+        mo_coeff_full = q_molecule.mo_coeff
+        mo_coeff_full_b = q_molecule.mo_coeff_b
+        # get molecular orbital occupation numbers
+        mo_occ_full = q_molecule.mo_occ
+        mo_occ_full_b = q_molecule.mo_occ_b
+        beta = mo_coeff_full_b is not None
+        mo_occ_total = mo_occ_full + mo_occ_full_b if beta else mo_occ_full
+        # compute number of inactive electrons
+        nelec_total = q_molecule.num_alpha + q_molecule.num_beta
+        nelec_inactive = nelec_total - self.num_electrons
+        if self.num_alpha is not None:
+            if not beta:
+                warning = 'The provided instance of QMolecule does not provide any beta ' \
+                          + 'coefficients but you tried to specify a separate number of alpha ' \
+                          + 'electrons. Continuing as if it does not matter.'
+                logger.warning(warning)
+            num_alpha = self.num_alpha
+            num_beta = self.num_electrons - self.num_alpha
+        else:
+            num_beta = (self.num_electrons - (q_molecule.multiplicity - 1)) // 2
+            num_alpha = self.num_electrons - num_beta
+        if nelec_inactive < 0:
+            raise QiskitNatureError("More electrons requested than available.")
+        if nelec_inactive % 2 != 0:
+            raise QiskitNatureError("The number of inactive electrons must be even.")
+
+        # determine active and inactive orbital indices
+        if self.active_orbitals is None:
+            norbs_inactive = nelec_inactive // 2
+            if norbs_inactive + self.num_orbitals > q_molecule.num_orbitals:
+                raise QiskitNatureError("More orbitals requested than available.")
+            inactive_orbs_idxs = list(range(norbs_inactive))
+            active_orbs_idxs = list(range(norbs_inactive, norbs_inactive+self.num_orbitals))
+        else:
+            if self.num_orbitals != len(self.active_orbitals):
```

Comment:
I would extract such checks to helper functions, e.g. `validate_num_orbitals()`.